PlayCover/Utils/NavigationStack.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  NavigationStack.swift
+//  PlayCover
+//
+//  Created by Amir Mohammadi on 11/2/1401 AP.
+//
+
+import SwiftUI
+
+enum StackNavigationTransition: Equatable {
```

Comment:
Is Equatable the best option to handle this operations?

Reply:
Actually yes, and those checks were suggested by KhoraLee before here https://github.com/PlayCover/PlayCover/pull/732